jbmc/src/java_bytecode/load_extra_methods.h

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************\
+
+Module: java bytecode
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+/// \file
+/// load_extra_methods for ci_lazy_methods with a specified list of methods
+
+#ifndef CPROVER_JAVA_BYTECODE_LOAD_EXTRA_METHODS_H
+#define CPROVER_JAVA_BYTECODE_LOAD_EXTRA_METHODS_H
+
+#include <vector>
+
+#include <util/irep.h>
+
+#include <java_bytecode/ci_lazy_methods.h>
+
+class symbol_tablet;
+
+class extra_methods_listt : public load_extra_methodst
+{
+public:
+  extra_methods_listt(
+    const std::vector<irep_idt> &lazy_methods_extra_entry_points)
+    : lazy_methods_extra_entry_points(lazy_methods_extra_entry_points)
+  {
+  }
+  std::vector<irep_idt>
+  extra_methods(const symbol_tablet &symbol_table) const override;
```

Comment:
This class hierarchy appears to be unnecessary. You could greatly simplify all of that by using lambdas (and maybe higher-order functions to make your life easier).